### PR TITLE
Use SmallRye BeanBag to initialize Maven RepositorySystem for compatibility with Maven 3.9

### LINKED ISSUE #31011: Impossible to create a project with Maven 3.9.0
### Describe the bug

When tring to bootstrap a Quarkus project with  this command 
`mvn com.redhat.quarkus.platform:quarkus-maven-plugin:2.13.5.SP1-redhat-00002:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=quarkus-backend \
    -DplatformGroupId=com.redhat.quarkus.platform \
    -DplatformVersion=2.13.5.SP1-redhat-00002 \
    -DclassName="org.mycompany.quarkusbackend" \
    -Dpath="/quarkus-backend" \
    -Dextensions="resteasy,openshift`

And keep having this errors : 
` A required class was missing while executing com.redhat.quarkus.platform:quarkus-maven-plugin:2.13.5.SP1-redhat-00002:create: org/eclipse/aether/connector/basic/BasicRepositoryConnectorFactory`

### Expected behavior

Bootstrap succeed

### Actual behavior

Errors with this message : ` A required class was missing while executing com.redhat.quarkus.platform:quarkus-maven-plugin:2.13.5.SP1-redhat-00002:create: org/eclipse/aether/connector/basic/BasicRepositoryConnectorFactory`

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Darwin mac 22.3.0 Darwin Kernel Version 22.3.0: Thu Jan  5 20:53:49 PST 2023; root:xnu-8792.81.2~2/RELEASE_X86_64 x86_64

### Output of `java -version`

java version "17.0.6" 2023-01-17 LTS Java(TM) SE Runtime Environment (build 17.0.6+9-LTS-190) Java HotSpot(TM) 64-Bit Server VM (build 17.0.6+9-LTS-190, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.0

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #31011

### CODE REVIEW

**Comment by aloubyansky** [comment]
Looks like a conflict in the assembly plugin. It did pass locally though.

**Review by dmlloyd — APPROVED** [review]
The BeanBag parts look good to me. Everything else seems to make sense but I'm not as familiar with some of this code.

File: independent-projects/bootstrap/pom.xml [hunk 1]
```diff
@@ -49,10 +49,10 @@
         <eclipse-minimal-json.version>0.9.5</eclipse-minimal-json.version>
         <jboss-logging.version>3.5.0.Final</jboss-logging.version>
         <junit.jupiter.version>5.9.2</junit.jupiter.version>
-        <maven-core.version>3.8.7</maven-core.version><!-- Keep in sync with sisu.version -->
+        <maven-core.version>3.9.0</maven-core.version><!-- Keep in sync with sisu.version -->
```
Comment (famod) [inline independent-projects/bootstrap/pom.xml:52]: Given that 3.9.1 is about to be released and contains some rather important fixed, I'd personally suggest to skip 3.9.0.

**Comment by famod** [comment]
What does that mean for #31836?

File: independent-projects/bootstrap/pom.xml (hunk 1 above)
Comment (aloubyansky) [reply independent-projects/bootstrap/pom.xml:52]: No problem skipping 3.9.0.

**Comment by aloubyansky** [comment]
Once this one passes and gets merged, I think it would make sense to update it to 3.9.x.

File: independent-projects/bootstrap/pom.xml [hunk 2]
```diff
@@ -71,7 +72,10 @@
         <slf4j-api.version>1.7.36</slf4j-api.version>
         <graal-sdk.version>22.3.0</graal-sdk.version>
         <plexus-classworlds.version>2.6.0</plexus-classworlds.version> <!-- not actually used but ClassRealm class is referenced from the API used in BootstrapWagonConfigurator -->
+        <plexus-cipher.version>1.7</plexus-cipher.version>
+        <plexus-utils.version>3.5.0</plexus-utils.version>
```
Comment (gnodet) [inline independent-projects/bootstrap/pom.xml:76]: Plexus-utils should be bumped to 3.5.1 at the same time.

**Comment by famod** [comment]
> Once this one passes and gets merged, I think it would make sense to update it to 3.9.x.

Ok, makes sense.

I'm wondering whether we need additional CI coverage for Maven 3.8? Then again we don't have coverage for 3.6 either. :thinking:

File: independent-projects/bootstrap/pom.xml (hunk 2 above)
Comment (aloubyansky) [reply independent-projects/bootstrap/pom.xml:76]: Thanks, updating.

**Comment by aloubyansky** [comment]
Apparently, the presence of the new dependencies of the bootstrap resolver on the classpath of a project may break the assembly plugin https://github.com/quarkusio/quarkus/blob/main/docs/pom.xml#L2981

File: independent-projects/bootstrap/maven-plugin/pom.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
             <plugin>
                 <groupId>org.codehaus.plexus</groupId>
                 <artifactId>plexus-component-metadata</artifactId>
-                <version>2.1.0</version>
+                <version>2.1.1</version>
```
Comment (cstamas) [inline independent-projects/bootstrap/maven-plugin/pom.xml:18]: 2.1.1 is fluke due due fluke qdox version. Go back to 2.1.0

As there is no newer version, while fix is committed https://github.com/codehaus-plexus/plexus-containers/commit/88f833bcc8ce843441fca1de669989c5ade366b1

Comment (aloubyansky) [reply independent-projects/bootstrap/maven-plugin/pom.xml:18]: Thanks!

File: devtools/maven/pom.xml [hunk 4]
```diff
@@ -190,6 +249,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-plugin-plugin</artifactId>
+                <version>3.7.1</version>
```
Comment (cstamas) [inline devtools/maven/pom.xml:252]: I'd just generally move for whole build to 3.8.1 version. The POMs build-parent, bootstrap-parent and extension-maven-plugin needs this added to pluginMgmt

File: independent-projects/bootstrap/maven-plugin/pom.xml (hunk 3 above)
Comment (cstamas) [reply independent-projects/bootstrap/maven-plugin/pom.xml:18]: Actually, any reason why plexus annos are used here? This project is already to be archived https://github.com/codehaus-plexus/plexus-containers/issues/59 w/ plexus-shim as drop in replacement (only to maintain backward compat, NOT to create NEW code with it).

Comment (cstamas) [reply independent-projects/bootstrap/maven-plugin/pom.xml:18]: Will prep a PR that gets rid of ancient plexus annos in favor of JSR330 as it is unrelated to this PR

Comment (aloubyansky) [reply independent-projects/bootstrap/maven-plugin/pom.xml:18]: Thanks @cstamas!

File: docs/pom.xml [hunk 5]
```diff
@@ -38,8 +39,17 @@
...
     <dependencies>
+        <!-- the httpcomponents dependencies are here to align dependencies with the assembly plugin -->
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpcore</artifactId>
+        </dependency>
```
Comment (aloubyansky) [inline docs/pom.xml:51]: @cstamas @gnodet I had to add these dependencies here to make the assembly plugin work. Isn't it an issue in Maven?

File: independent-projects/bootstrap/pom.xml (hunk 1 above)
Comment (cstamas) [reply independent-projects/bootstrap/pom.xml:52]: 3.9.1 is out, https://maven.apache.org/docs/3.9.1/release-notes.html

File: docs/pom.xml (hunk 5 above)
Comment (cstamas) [reply docs/pom.xml:51]: I never saw problem like this in Maven. This may be related to invoking Mojos within Mojos? (just guessing here).

Comment (aloubyansky) [reply docs/pom.xml:51]: I'll have a look but it seems to fail on the assembly plugin in this case. If I comment it out, it builds.

Comment (aloubyansky) [reply docs/pom.xml:51]: Now I seem to be getting it for the jar plugin
```
[INFO] --- maven-jar-plugin:3.1.2:jar (default-jar) @ quarkus-documentation ---
Downloading from jboss-public-repository: https://repository.jboss.org/nexus/content/groups/public/org/codehaus/plexus/plexus-utils/1.4.2/plexus-utils-1.4.2.pom

Exception in thread "main" java.lang.NoSuchMethodError: org.apache.http.client.utils.URLEncodedUtils.formatSegments([Ljava/lang/String;)Ljava/lang/String;
	at org.apache.maven.wagon.shared.http.EncodingUtil.encodeURLToString(EncodingUtil.java:118)
	at org.apache.maven.wagon.shared.http.EncodingUtil.encodeURLToString(EncodingUtil.java:95)
	at org.apache.maven.wagon.providers.http.LightweightHttpWagon.buildUrl(LightweightHttpWagon.java:108)
	at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:116)
	at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:126)
	at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)
	at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
	at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run(WagonTransporter.java:433)
	at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:366)
	at org.eclipse.aether.transport.wagon.WagonTransporter.get(WagonTransporter.java:347)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask(BasicRepositoryConnector.java:457)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:364)
	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:75)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:628)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:262)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:514)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:402)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:229)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:207)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:240)
	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:171)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.resolveCachedArtifactDescriptor(DefaultDependencyCollector.java:538)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.getArtifactDescriptorResult(DefaultDependencyCollector.java:523)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:410)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:362)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:349)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.doRecurse(DefaultDependencyCollector.java:506)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:458)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:362)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:349)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.doRecurse(DefaultDependencyCollector.java:506)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:458)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:362)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:349)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.doRecurse(DefaultDependencyCollector.java:506)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:458)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:362)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:349)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.doRecurse(DefaultDependencyCollector.java:506)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:458)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.processDependency(DefaultDependencyCollector.java:362)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.process(DefaultDependencyCollector.java:349)
	at org.eclipse.aether.internal.impl.collect.DefaultDependencyCollector.collectDependencies(DefaultDependencyCollector.java:254)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.collectDependencies(DefaultRepositorySystem.java:284)
	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveInternal(DefaultPluginDependenciesResolver.java:218)
	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve(DefaultPluginDependenciesResolver.java:165)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.createPluginRealm(DefaultMavenPluginManager.java:404)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.setupPluginRealm(DefaultMavenPluginManager.java:374)
	at org.apache.maven.plugin.DefaultBuildPluginManager.getPluginRealm(DefaultBuildPluginManager.java:234)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:105)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

**Review by dmlloyd — APPROVED** [review]
One comment, otherwise looks good to me

File: devtools/maven/src/main/java/io/quarkus/maven/components/QuarkusWorkspaceProvider.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+    VersionRangeResolver versionRangeResolver;
+
+    ArtifactResolver artifactResolver;
+
+    MetadataResolver metadataResolver;
+
+    Deployer deployer;
+
+    RemoteRepositoryManager remoteRepoManager;
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/components/QuarkusWorkspaceProvider.java:42]: Due to https://github.com/quarkusio/quarkus/issues/32081

**Comment by aloubyansky** [comment]
These failures aren't related to the changes in this PR.

**Review by gsmet — APPROVED** [review]

**Pull request merged**